src/lib/mod.rs

```diff
@@ -1,14 +1,72 @@
 pub mod barcode_matching;
 pub mod samples;
 
+use bio::data_structures::bitenc::BitEnc;
+use lazy_static::lazy_static;
+
+pub const DNA_BASES: [u8; 5] = *b"ACGTN";
+pub const IUPAC_BASES: [u8; 15] = *b"ACGTMRWSYKVHDBN";
+
+lazy_static! {
+    pub static ref BASE_A: usize = 1;
+    pub static ref BASE_C: usize = 2;
+    pub static ref BASE_G: usize = 4;
+    pub static ref BASE_T: usize = 8;
+    pub static ref BASE_N: usize = 15;
+    pub static ref DNA_MASKS: [u8; 256] = {
+        let mut masks = [0; 256];
+        let (a, c, g, t) = (1, 2, 4, 8);
+        masks['A' as usize] = a;
+        masks['C' as usize] = c;
+        masks['G' as usize] = g;
+        masks['T' as usize] = t;
+        masks['U' as usize] = t;
+        masks['N' as usize] = a | c | g | t;
+        masks
+    };
+    pub static ref IUPAC_MASKS: [u8; 256] = {
+        let mut masks = [0; 256];
+        let (a, c, g, t) = (1, 2, 4, 8);
+        masks['A' as usize] = a;
+        masks['C' as usize] = c;
+        masks['G' as usize] = g;
+        masks['T' as usize] = t;
+        masks['U' as usize] = t;
+        masks['M' as usize] = a | c;
+        masks['R' as usize] = a | g;
+        masks['W' as usize] = a | t;
+        masks['S' as usize] = c | g;
+        masks['Y' as usize] = c | t;
+        masks['K' as usize] = g | t;
+        masks['V' as usize] = a | c | g;
+        masks['H' as usize] = a | c | t;
+        masks['D' as usize] = a | g | t;
+        masks['B' as usize] = c | g | t;
+        masks['N' as usize] = a | c | g | t;
+        masks
+    };
+}
+
+#[must_use]
+pub fn encode(bases: &[u8]) -> BitEnc {
+    let mut vec = BitEnc::with_capacity(4, bases.len());
```

Comment:
`bio` is fairly heavy, and `BitEnc` is probably not the fastest implementation of bit encoding. This is a good resource if you want to roll your own: https://github.com/Daniel-Liu-c0deb0t/cute-nucleotides

Some other crates to check out:
* https://github.com/noamteyssier/bitnuc
* https://github.com/jeff-k/bio-seq

Reply:
`bitnuc` is 2-bit encoded, whereas we need 4 bits for IUPAC codes, which it looks like bio-seq does!